oci-devfile-registry-metadata/nginx.conf

```diff
@@ -0,0 +1,62 @@
+events {
+    worker_connections  1024;
+}
+
+http {
+
+  upstream docker-registry {
+    server localhost:5000;
+  }
+
+  ## Set a variable to help us decide if we need to add the
+  ## 'Docker-Distribution-Api-Version' header.
+  ## The registry always sets this header.
+  ## In the case of nginx performing auth, the header is unset
+  ## since nginx is auth-ing before proxying.
+  map $upstream_http_docker_distribution_api_version $docker_distribution_api_version {
+    '' 'registry/2.0';
+  }
+
+  server {
+    listen 8080;
+    
+    # Recommendations from https://raymii.org/s/tutorials/Strong_SSL_Security_On_nginx.html
+    ssl_protocols TLSv1.1 TLSv1.2;
+    ssl_ciphers 'EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH';
+    ssl_prefer_server_ciphers on;
+    ssl_session_cache shared:SSL:10m;
+
+    # disable any limits to avoid HTTP 413 for large image uploads
+    client_max_body_size 0;
+
+    # required to avoid HTTP 411: see Issue #1486 (https://github.com/moby/moby/issues/1486)
+    chunked_transfer_encoding on;
+
+    location / {
+        # Temporary until the new metadata server has been delivered
+        # ToDo: Update to point to the new server when ready
+        default_type application/json;
+        root /www/data;
+        index index.json;
+    }
+
+    location /v2 {      
+      # Only allow HEAD (implicit) and GET requests on the V2 api
+      limit_except GET {
```

Comment:
Please correct me if I am wrong. Seems like the current nginx server works as reverse proxy and redirects:
- registry API requests to registry container (block all requests except GET requests)
- other requests to index server

Given our current index server already exposes the REST APIs (/index, /devfiles/{name}) to consumers to let consumers to GET index and devfile, in other words, it already implicitly blocks the PUT/POST/DELETE requests as we don't support them for now, for the time being, seems like the current index server has the same functionalities as using index server with nginx.

Also I am thinking more about the future work for supporting dynamic registry (support POST/PUT/DELETE) as I believe currently both index server and index server with nginx don't support those functionalities.
- If we use index server we can expose more REST APIs and design it to make it more user-friendly for devfile user, inside index server we still use ORAS package to interact with registry container (it's truly not accessible from public)
- If we use index server with nginx, we rely on registry API to let use interact with registry container directly within index server container, for this case I am not sure if it's easier to use as the GET registry API is `GET /v2/<name>/blobs/<digest>`, seems like user still need to pass/store some other information instead of `GET /devfiles/{name}`

Reply:
You're right, it is a reverse proxy, and the doc here: https://docs.docker.com/registry/recipes/nginx/ confirms that. I'll update the PR description.

The reason we still need to expose the OCI registry to clients is because more than just the devfiles will be stored in the OCI registry (e.g. plugins, stack artifacts, etc). Using the index server's `GET /devfiles/{name}` is good for retrieving the devfile, but it doesn't work well if other artifacts are stored as part of the stack (like binary blobs).